src/test/java/space/obminyashka/items_exchange/controller/UserControllerTest.java

```diff
@@ -78,116 +81,57 @@ void negativeTestReceivingInformationAboutAnotherUser() throws Exception {
     }
 
     @Test
     @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn400WhenFirstNameContainsTwoWords() throws Exception {
-        getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithInvalidFNameWithoutPhones(), status().isBadRequest())
-                .andReturn();
-    }
+    void updateUserInfo_invalidFirstAndLastName_shouldReturnHttpStatusBadRequest() throws Exception {
+        UserUpdateDto dto = createUserUpdateDtoWithInvalidFirstAndLastName();
 
-    @Test
-    @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn400WhenDuplicateEmail() throws Exception {
-        when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
-        when(userService.existsByEmail(any())).thenReturn(true);
+        final var errorMessageForInvalidFirstName = getMessageSource("invalid.first-or-last.name")
+                .replace("${validatedValue}", dto.getFirstName());
+        final var errorMessageForInvalidLastName = getMessageSource("invalid.first-or-last.name")
+                .replace("${validatedValue}", dto.getLastName());
 
-        MvcResult mvcResult = getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithDuplicateEmailWithoutPhones(), status().isBadRequest())
-                .andDo(print())
-                .andReturn();
-        String message = Objects.requireNonNull(mvcResult.getResolvedException()).getMessage();
-
-        assertTrue(message.contains(getMessageSource("email.duplicate")));
-    }
-
-    @Test
-    @WithMockUser(username = "deletedUser")
-    void updateUserInfo_WhenUserHasStatusDeleted_ShouldReturn403WithSpecificMessage() throws Exception {
-        user.setStatus(Status.DELETED);
-        when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
-        when(userService.getDaysBeforeDeletion(any())).thenReturn(7L);
-
-        MvcResult mvcResult = getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithChangedEmailAndFNameApAndLNameMinusWithoutPhones(),
-                status().isForbidden())
+        MvcResult mvcResult = getResultActions(HttpMethod.PUT, "/user/info", dto, status().isBadRequest())
                 .andDo(print())
                 .andReturn();
-
-        User deletedUser = userService.findByUsernameOrEmail("deletedUser@gmail.com").orElseThrow();
-        String contentAsString = mvcResult.getResponse().getContentAsString();
-
-        assertTrue(contentAsString.contains(getMessageSource("exception.illegal.operation")
-                .concat(". ")
-                .concat(getParametrizedMessageSource("account.self.delete.request",
-                        userService.getDaysBeforeDeletion(deletedUser)))));
+        Assertions.assertAll(
+                () ->  assertTrue(mvcResult.getResponse().getContentAsString().contains(errorMessageForInvalidFirstName)),
```

Comment:
`mvcResult.getResponse().getContentAsString()` can be extracted into variable for both cases.

Reply:
ok, but this logic duplicates in another my tests, so i`ve  extracted this into the separate method. After that, can i refactor all  methods where using this logic and  i`m not owner, and replace this logic to the new method ?